GameServerLib/GameObjects/AttackableUnits/AI/ObjAIBase.cs

```diff
@@ -455,65 +455,37 @@ float travelTime
         /// TODO: Re-implement this for LaneMinions and add a patience or distance threshold so they don't follow forever.
         public bool RecalculateAttackPosition()
         {
-            // If we are already where we should be, which means we are in attack range and not colliding, then keep our current position.
-            if (TargetUnit != null && !TargetUnit.IsDead && Vector2.DistanceSquared(Position, TargetUnit.Position) > CollisionRadius * CollisionRadius
-                && Vector2.DistanceSquared(Position, TargetUnit.Position) <= Stats.Range.Total * Stats.Range.Total)
+            // If we are already where we should be, which means we are in attack range, then keep our current position.
+            if (TargetUnit != null && !TargetUnit.IsDead && Vector2.DistanceSquared(Position, TargetUnit.Position) <= Stats.Range.Total * Stats.Range.Total)
             {
                 return false;
             }
-            var objects = _game.ObjectManager.GetObjects();
-            List<CirclePoly> usedPositions = new List<CirclePoly>();
-            var isCurrentlyOverlapping = false;
 
-            var thisCollisionCircle = new CirclePoly(TargetUnit?.Position ?? Position, CollisionRadius + 10);
+            var nearestObjects = _game.Map.CollisionHandler.QuadDynamic.GetNearestObjects(this);
 
-            foreach (var gameObject in objects)
+            foreach (var gameObject in nearestObjects)
             {
-                var unit = gameObject.Value as IAttackableUnit;
+                var unit = gameObject as IAttackableUnit;
                 if (unit == null ||
                     unit.NetId == NetId ||
                     unit.IsDead ||
-                    unit.Team != Team ||
```

Comment:
What is the reason for this condition being deleted?

Reply:
Collisions are not team specific.